### PR TITLE
AbstractHttp2ConnectionHandlerBuilder validateHeaders cannot be set w…

### PR DESCRIPTION
**Pull request by Scottmitch**
…ith encoder/decoder

Motivation:
If validateHeaders is set in combination with the encoder/decoder it will be silently ignored. We should enforce the constraint that validateHeaders and encoder/decoder are mutually exclusive.

Modifications:
- Make sure either validateHeaders can be set or encoder/decoder.

Result:
AbstractHttp2ConnectionHandlerBuilder does not allow conflicting options to be set.

### CODE REVIEW

**Comment by Scottmitch** [comment]
@nmittler @trustin - PTAL

**Comment by normanmaurer** [comment]
@Scottmitch LGTM

File: codec-http2/src/main/java/io/netty/handler/codec/http2/AbstractHttp2ConnectionHandlerBuilder.java [hunk 1]
```diff
@@ -246,6 +230,24 @@ protected B codec(Http2ConnectionDecoder decoder, Http2ConnectionEncoder encoder
...
+    protected boolean isValidateHeaders() {
+        return validateHeaders != null ? validateHeaders : true;
+    }
+
+    /**
+     * Sets if HTTP headers should be validated according to
+     * <a href="https://tools.ietf.org/html/rfc7540#section-8.1.2.6">RFC 7540, 8.1.2.6</a>.
+     */
+    protected B validateHeaders(boolean validateHeaders) {
+        enforceNonCodecConstraints("validateHeaders");
```
Comment (nmittler) [inline codec-http2/src/main/java/io/netty/handler/codec/http2/AbstractHttp2ConnectionHandlerBuilder.java]: Could we come up with a better term than `enforceNonCodecConstraints`? I had to look at the method to see what it does.  Maybe `checkNotBuilt` or something to that effect?

Comment (nmittler) [inline codec-http2/src/main/java/io/netty/handler/codec/http2/AbstractHttp2ConnectionHandlerBuilder.java]: similarly, could we change `enforceConstraint` to something like `checkFieldNotSet`?

**Comment by nmittler** [comment]
@Scottmitch LGTM ... just a couple of nits.

File: codec-http2/src/main/java/io/netty/handler/codec/http2/AbstractHttp2ConnectionHandlerBuilder.java (hunk 1 above)
Comment (trustin) [inline codec-http2/src/main/java/io/netty/handler/codec/http2/AbstractHttp2ConnectionHandlerBuilder.java]: Couldn't find the good names for those methods, but `checkNotBuild/checkFieldNotSet` doesn't sound enough to me. Hmm..

**Comment by trustin** [comment]
LGTM

File: codec-http2/src/main/java/io/netty/handler/codec/http2/AbstractHttp2ConnectionHandlerBuilder.java (hunk 1 above)
Comment (Scottmitch) [inline codec-http2/src/main/java/io/netty/handler/codec/http2/AbstractHttp2ConnectionHandlerBuilder.java]: @nmittler - I'm not apposed to it but lets discuss the rename in a followup PR.

**Comment by Scottmitch** [comment]
Cherry-picked 4.1 (78b508a)

**Pull request closed without merging**